mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -340,6 +340,23 @@ def lookup(self, kwargs: TYPE_nkwargs, force_fallback: bool = False) -> Dependen
                         self._subproject_impl(subp_name, varname)
                     break
 
+                # Check if we could have a matching Rust crate imported from
+                # Cargo.lock, with expected api version. We know there is no
+                # wrap-file defined to match this crate.
+                if name.endswith('-rs') and required:
+                    s = name.split('-')
+                    if len(s) < 3:
+                        continue
+                    s.pop() # -rs
+                    s.pop() # version
+                    crate = '-'.join(s)
```

Comment:
I'm not sure whether this block of code to extract the crate name and version is really necessary. It seems that all it tries to do is to superficially validate that the dependency name is in the form `$name-$version-rs`. However, it does not enforce that `$version` is a valid version string. What I would have done is simply:
```python
if name.endswith('-rs') and required:
   cargo_wraps = self.wrap_resolver.get_cargo_wraps(name[:-3])
   ...
```
I assume that looking up the wrap is relatively cheap and, because this is anyhow only an error path, filtering out false positives (dependency names that look like Rust crates but are not) is not really worth the extra code.

Reply:
This code extract the "correct" name of the create (crate names can contain `-`, thus the manual decomposition).
In more, we want to filter out the version, precisely to search other versions available.
`get_cargo_wraps` takes the crate name, without version included.

Reply:
There were already lint errors due to popping multiple members of this list. I think it's a sign that the correct solution for implementing the existing logic was actually `name.rsplit(maxsplit=2)[0]`.

Reply:
I fixed the proposed code.

Reply:
Using the proposed change.